File: src/main/java/me/zeroeightsix/kami/event/events/ScreenEvent.java
package me.zeroeightsix.kami.event.events;

import me.zeroeightsix.kami.event.KamiEvent;
import net.minecraft.client.gui.screen.Screen;

/**
 * Created by 086 on 17/11/2017.
 */
public class ScreenEvent extends KamiEvent  {

    private Screen screen;

    public ScreenEvent(Screen screen) {
        super();
        this.screen = screen;
    }

    public Screen getScreen() {
        return screen;
    }

    public void setScreen(Screen screen) {
        this.screen = screen;
    }

    public static class Displayed extends ScreenEvent {
        public Displayed(Screen screen) {
            super(screen);
        }
    }

    public static class Closed extends ScreenEvent {
        public Closed(Screen screen) {
            super(screen);
        }
    }

}

File: src/main/kotlin/me/zeroeightsix/kami/event/events/ScreenEvent.kt
package me.zeroeightsix.kami.event.events

import me.zeroeightsix.kami.event.KamiEvent
import net.minecraft.client.gui.screen.Screen

/**
 * Created by 086 on 17/11/2017.
 */
open class ScreenEvent(var screen: Screen?) : KamiEvent() {

    class Displayed(screen: Screen?) : ScreenEvent(screen)
    class Closed(screen: Screen?) : ScreenEvent(screen)

}